tests/macros/utils.js

```diff
@@ -63,15 +65,37 @@ function createMacroTestObject(macroName) {
         url: 'https://developer.mozilla.org/'
     };
     let environment = new Environment(pageContext, templates, true);
+    let ctx = environment.prototypeEnvironment;
+
+    /** @type {Map<string, string>} */
+    const macroResults = new Map();
+    const realTemplate = ctx.template;
+    ctx.template = jest.fn(async (name, ...args) => {
+        let macroName = String(name)
+            .replace(/:/g, '-')
+            .toLowerCase();
+        return macroResults.get(macroName) || realTemplate(name, ...args);
+    });
 
     return {
         /**
          * Give the test-case writer access to the macro's globals (ctx).
          * For example, "macro.ctx.env.locale" can be manipulated to something
          * other than 'en-US' or "macro.ctx.wiki.getPage" can be mocked
-         * using "sinon.stub()" to avoid network calls.
+         * using `jest.fn()` to avoid network calls.
+         */
+        ctx,
+
+        /**
+         * A much easier way to handle overwriting the result of the `template`
+         * function for specific macro calls.
```

Comment:
Could you expand on this comment, please? Something like:

"When writing tests for a macro that invokes other macros with the template() function
you sometimes want to specify a mock return value for those other macros. This function..."

And also in your comment you might mention that you can turn off a mock by calling the function again and passing null as the result.